asv_bench/benchmarks/categoricals.py

```diff
@@ -1,114 +1,140 @@
-from .pandas_vb_common import *
-try:
-    from pandas.api.types import union_categoricals
-except ImportError:
-    try:
-        from pandas.types.concat import union_categoricals
-    except ImportError:
-        pass
+import numpy as np
+import pandas as pd
+import pandas.util.testing as tm
+from pandas.core.dtypes.concat import union_categoricals
 
 
-class Categoricals(object):
+class Concat(object):
+
     goal_time = 0.2
 
     def setup(self):
-        N = 100000
-        self.s = pd.Series((list('aabbcd') * N)).astype('category')
+        N = 10**5
+        self.s = pd.Series(list('aabbcd') * N).astype('category')
+
+        self.a = pd.Categorical(list('aabbcd') * N)
+        self.b = pd.Categorical(list('bbcdjk') * N)
+
+    def time_concat(self):
+        pd.concat([self.s, self.s])
+
+    def time_union(self):
+        union_categoricals([self.a, self.b])
+
 
-        self.a = pd.Categorical((list('aabbcd') * N))
-        self.b = pd.Categorical((list('bbcdjk') * N))
+class Constructor(object):
 
+    goal_time = 0.2
+
+    def setup(self):
+        N = 10**5
         self.categories = list('abcde')
-        self.cat_idx = Index(self.categories)
+        self.cat_idx = pd.Index(self.categories)
         self.values = np.tile(self.categories, N)
         self.codes = np.tile(range(len(self.categories)), N)
 
-        self.datetimes = pd.Series(pd.date_range(
-            '1995-01-01 00:00:00', periods=10000, freq='s'))
+        self.datetimes = pd.Series(pd.date_range('1995-01-01 00:00:00',
+                                                 periods=N / 10,
+                                                 freq='s'))
+        self.datetimes_with_nat = self.datetimes.copy()
+        self.datetimes_with_nat.iloc[-1] = pd.NaT
 
         self.values_some_nan = list(np.tile(self.categories + [np.nan], N))
         self.values_all_nan = [np.nan] * len(self.values)
 
-    def time_concat(self):
-        concat([self.s, self.s])
-
-    def time_union(self):
-        union_categoricals([self.a, self.b])
-
     def time_constructor_regular(self):
```

Comment:
Do we wan to remove the `_constructor` part in the time names? As you already have it in the class name (now you have 'constructor' twice and making the name long)